kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/framework/concurrency/Interval.kt

```diff
@@ -0,0 +1,113 @@
+package io.kotest.framework.concurrency
+
+import io.kotest.common.ExperimentalKotest
+import kotlin.math.pow
+
+private const val hour = 3_600_000L
+
+/**
+ * A [Interval] determines how often Kotest will invoke the predicate function for an [until] block.
+ */
+@ExperimentalKotest
+interface Interval {
+
+   /**
+    * Returns the next delay as a [Millis].
+    *
+    * @param count The number of times the condition has been polled (evaluated) so far.
+    * Always a positive integer.
+    *
+    * @return The duration of the next poll interval
+    */
+   fun next(count: Int): Millis
```

Comment:
So this is not a suggestion but a discussion point - if you make next() return a Pair<Interval, Millis> then you have no state to worry about anywhere, as the count can be internalized.

Reply:
Ooo I like this, I'll play around with that idea 👍